packages/shared/src/hooks/referral/useReferralCampaign.ts

```diff
@@ -0,0 +1,115 @@
+import { useQuery } from 'react-query';
+import { useContext } from 'react';
+import { useRequestProtocol } from '../useRequestProtocol';
+import { REFERRAL_CAMPAIGN_QUERY } from '../../graphql/users';
+import { graphqlUrl } from '../../lib/config';
+import { RequestKey, generateQueryKey } from '../../lib/query';
+import AuthContext from '../../contexts/AuthContext';
+import { feature, Feature } from '../../lib/featureManagement';
+import { useFeatureIsOn } from '../../components/GrowthBookProvider';
+
+export interface ReferralCampaign {
+  referredUsersCount: number;
+  referralCountLimit: number;
+  url: string;
+}
+
+export interface UseReferralCampaign extends ReferralCampaign {
+  isCompleted: boolean;
+  isReady: boolean;
+  availableCount: number;
+  noKeysAvailable: boolean;
+}
+
+export enum ReferralCampaignKey {
+  Search = 'search',
+}
+
+export type UseReferralCampaignProps = {
+  campaignKey: ReferralCampaignKey;
+};
+
+export const campaignToReferralTargetCountMap: Record<
+  ReferralCampaignKey,
+  number
+> = { search: 5 };
+
+const campaignFeatureFlagMap: Partial<
+  Record<ReferralCampaignKey, Feature<string>>
+> = {
+  search: feature.search,
+};
```

Comment:
This should also come from the backend I think.

Reply:
We need it here to identify which feature is for what campaign. This is also to understand whether they have access to it already, and if so we redirect.